archinfo/arch_riscv.py

```diff
@@ -0,0 +1,164 @@
+try:
+    import capstone as _capstone
+    if _capstone.__version__ < '5.0':
+        raise ImportError('Only capstone verison >= 5.0 support RISC-V')
+except ImportError:
+    _capstone = None
+
+from .arch import Arch, register_arch, Endness, Register
+from .archerror import ArchError
+from .tls import TLSArchInfo
+
+
+class ArchRISCV(Arch):
+    def __init__(self, endness=Endness.LE):
+        if endness != Endness.LE:
+            raise ArchError('Arch RISCV must be little endian')
+        super(ArchRISCV, self).__init__(endness)
+
+    address_types = (int,)
+    function_address_types = (int,)
+
+    name = 'RISCV'
+    bits = 32
+    vex_arch = None  # No VEX support
+    qemu_name = None  # No Unicorn-engine support
+    ida_processor = None
+    triplet = 'riscv32-linux-gnu'
+    max_inst_bytes = 4
+
+    ip_offset = None
+    sp_offset = None
+    bp_offset = None
+    lr_offset = None
+    ret_offset = 11  # a0(x10) For return value
```

Comment:
the meaning of "offset" here is the vex offset into the register file. 11 is definitely not the offset of a register... same with `syscall_num_offset` below. Also at the very least you should definitely fill in `ip_offset`, if not the rest as well.